### ISSUE TITLE
Assembler with incorrect register errors, building linux.

### ISSUE THREAD

**Issue by TimJSwan89** [gwsystems/ps#2]
Build issues.
`git clone https://github.com/gwsystems/ps.git`
```
VirtualBox:~/Desktop/ps$ ./configure
Usage: ./configure osname archname
	osname is in {linux, cos}
	archname is in {x86}
VirtualBox:~/Desktop/ps$ ./configure linux x86
VirtualBox:~/Desktop/ps$ make
gcc -MD -MP -std=c11 -Wall -Wpedantic -Wextra -Wno-long-long -O3 -I.  -Wno-unused-function -o ps_smr.o -c ps_smr.c
gcc -MD -MP -std=c11 -Wall -Wpedantic -Wextra -Wno-long-long -O3 -I.  -Wno-unused-function -o ps_ns.o -c ps_ns.c
./plat/arch/x86/ps_arch.h: Assembler messages:
./plat/arch/x86/ps_arch.h:71: Error: incorrect register `%rsi' used with `l' suffix
./plat/arch/x86/ps_arch.h:71: Error: incorrect register `%r14' used with `l' suffix
./plat/arch/x86/ps_arch.h:81: Error: incorrect register `%r13' used with `l' suffix
./plat/arch/x86/ps_arch.h:71: Error: incorrect register `%rsi' used with `l' suffix
./plat/arch/x86/ps_arch.h:81: Error: incorrect register `%rax' used with `l' suffix
Makefile:26: recipe for target 'ps_ns.o' failed
make: *** [ps_ns.o] Error 1
```

Perhaps unrelated, but more information:
```
VirtualBox:~/Desktop/ps$ ./configure cos x86
VirtualBox:~/Desktop/ps$ make
gcc -MD -MP -std=c11 -Wall -Wpedantic -Wextra -Wno-long-long -O3 -I.  -Wno-unused-function -o ps_ns.o -c ps_ns.c
In file included from ./ps_plat.h:4:0,
                 from ./ps_global.h:11,
                 from ./ps_smr.h:29,
                 from ./ps_ns.h:22,
                 from ps_ns.c:8:
./plat/os/cos/ps_os.h:15:23: fatal error: cos_debug.h: No such file or directory
compilation terminated.
Makefile:26: recipe for target 'ps_ns.o' failed
make: *** [ps_ns.o] Error 1
```

**Comment by gparmer** [comment]
This is brilliant.  Thanks for the report @TimJSwan89.  It looks like a few things are happening with 32 bit vs. 64 bit issues, and using Composite includes rather than Linux ones.  Sorry, but we do a lot of testing in Composite, so we likely didn't notice these problems.

I'll try and figure these out in the next few days.  Thanks for the feedback.  Can you post the output for `cat /proc/cpuinfo`, and `gcc --version`, and the version of the Linux distro you're using?

**Comment by gparmer** [comment]
@ryuxin if you have the time to look at this.

**Comment by ryuxin** [comment]
@TimJSwan89 
If you use 64 bit system, you need to change ```PS_WORDSIZE``` to 64
https://github.com/gwsystems/ps/blob/master/ps_config.h#L50

**Comment by gparmer** [comment]
@ryuxin we should figure out a way to make this automatically configured, or at least erroring out if it is mis-configured.  Thanks for looking into this!

**Comment by gparmer** [comment]
Solved in #4 

Added a new architecture `x86_64`.
